src/MycilaConfig.cpp

```diff
@@ -95,45 +121,35 @@ bool Mycila::Config::set(const char* key, std::string value, bool fireChangeCall
   return op == Op::UNSET;
 }
 
-Mycila::Config::Op Mycila::Config::_set(const char* key, const char* value, bool fireChangeCallback) {
-  const bool del = value == nullptr || !value[0];
-
+Mycila::Config::Op Mycila::Config::_set(const char* key, const std::string& value, bool fireChangeCallback) {
   // check if the key is valid
-  if (std::find(_keys.begin(), _keys.end(), key) == _keys.end()) {
-    if (del) {
-      LOGW(TAG, "unset(%s): Unknown key!", key);
-    } else {
-      LOGW(TAG, "set(%s, %s): Unknown key!", key, value);
-    }
+  if (!exists(key)) {
+    LOGW(TAG, "set(%s, %s): Unknown key!", key, value.c_str());
     return Op::NOOP;
   }
 
-  // requested deletion ?
-  if (del) {
-    // key not there or not removed
-    if (!_prefs.isKey(key) || !_prefs.remove(key))
-      return Op::NOOP;
-
-    // key there and to remove
-    _cache.erase(key);
-    LOGD(TAG, "unset(%s)", key);
-    if (fireChangeCallback && _changeCallback)
-      _changeCallback(key, empty);
-    return Op::UNSET;
-  }
```

Comment:
where this logic went ?

Reply:
Ah ! My bad lol this was dead code! 
I did a refactor once to remove support for const char* values but did not remove this part!